src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1110,11 +1110,40 @@ bool entitiesQuery
     }
 
     /* Process query result */
-    while (cursor->more()) {
+    while (cursor->more())
+    {
+        BSONObj                 r    = cursor->next();
+        std::string             err  = r.getStringField("$err");
+        ContextElementResponse* cer  = new ContextElementResponse();
+
+        if (err != "")
+        {
+          //
+          // We can't return the error 'as is', as it may contain forbidden characters.
+          // So, we can just match the error and sent a less descriptive text
+          //
+          const char* invalidPolygon      = "Exterior shell of polygon is invalid";
+          const char* defaultErrorString  = "Internal mongo problem";
+
+          LM_W(("Database Error (%s)", err.c_str()));
+
+          if (strncmp(err.c_str(), invalidPolygon, strlen(invalidPolygon)) == 0)
+            err = invalidPolygon + std::string(" - not detected at creation time");
+          else
+            err = defaultErrorString;
+
+          //
+          // FIXME P1: It would be nice to fill in the entity but it is
+          //           difficult to do this, I simply fill it in with enV[0] ...
+          //
+          cer->contextElement.entityId.fill(enV[0]);
```

Comment:
We are inside entitiesQuery, so it is not clear which ngsi response is in question.
We can always return an empty Entity.

However, if there is *only one* entity in the request, we can return that one (as is done right now) and if there is more than one, we return an empty entity.

With an empty entity, I mean an entity with id == "", type == "", isPattern == "".